docs/subnets/create-a-vm-blobvm.md

```diff
@@ -0,0 +1,854 @@
+
+	return nil
+}
+```
+
+### Service
+
+[Service](https://github.com/ava-labs/blobvm/blob/master/vm/public_service.go) implements the API handlers to access the functions of the VM. VM has a method called [`CreateHandlers()`](https://github.com/ava-labs/blobvm/blob/master/vm/vm.go#L265) that will return the HTTP handler.
+
+```go
+func (vm *VM) CreateHandlers() (map[string]*common.HTTPHandler, error) {
+    apis := map[string]*common.HTTPHandler{}
+    public, err := newHandler(Name, &PublicService{vm: vm})
+    if err != nil {
+        return nil, err
+    }
+    apis[PublicEndpoint] = public
+    return apis, nil
+}
+```
+
+In the above function, a new RPC server is created with the `PublicService`. It implements all the RPCs supported by the virtual machine. It has access to a VM instance in its structure, that is passed while creating it (see the above function).
+
+```go
+type PublicService struct {
+	vm *VM
+}
+```
+
+Let's have a look at the [`IssueRawTx()`](https://github.com/ava-labs/blobvm/blob/master/vm/public_service.go#L63) service method -
+
+```go
+func (svc *PublicService) IssueRawTx(_ *http.Request, args *IssueRawTxArgs, reply *IssueRawTxReply) error {
+	tx := new(chain.Transaction)
+	if _, err := chain.Unmarshal(args.Tx, tx); err != nil {
+		return err
+	}
+
+	// otherwise, unexported tx.id field is empty
+	if err := tx.Init(svc.vm.genesis); err != nil {
+		return err
+	}
+	reply.TxID = tx.ID()
+
+	errs := svc.vm.Submit(tx)
+	if len(errs) == 0 {
+		return nil
+	}
+	if len(errs) == 1 {
+		return errs[0]
+	}
+	return fmt.Errorf("%v", errs)
+}
+```
+It accepts `IssueRawTxArgs` that contain transaction bytes. It processes the request as following
+
+- Unmarshal transaction bytes into VM-defined transaction object
+- Initialize transaction object with txID, digest hash, etc.
+- Submit transaction object to VM.
+
+Similarly, all other services are implemented.
+
+### Mempool
+
+[Mempool](https://github.com/ava-labs/blobvm/blob/master/mempool/mempool.go) is temporary memory for storing pending transactions. These are maintained at the local node level and can be flushed out when the node restarts. The new transactions received directly from a client or through app gossip, are stored inside this mempool.
+
+Mempool is ideally a Max Heap that pushes new transactions into the heap according to the transaction's price. It keeps the transaction with the highest price at the top. Due to separate implementation of transaction heap and other complexities it has its own `mempool` directory.
+
+Mempool is [initialized](https://github.com/ava-labs/blobvm/blob/master/vm/vm.go#L151) during the initialization of VM.
+
+```go
+vm.mempool = mempool.New(vm.genesis, vm.config.MempoolSize)
+```
+
+Whenever a transaction is submitted to VM, it first gets initialized, verified, and executed locally. If the transaction looks valid, then it is [added](https://github.com/ava-labs/blobvm/blob/master/vm/vm.go#L431) to mempool.
+
+```go
+vm.mempool.Add(tx)
+```
```

Comment:
MD031/blanks-around-fences: Fenced code blocks should be surrounded by blank linesmarkdownlint[MD031](https://github.com/DavidAnson/markdownlint/blob/v0.25.1/doc/Rules.md#md031)